apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -1,31 +1,48 @@
 import React from 'react';
 import color from '@cdo/apps/util/color';
 
+const spriteLocation = {
+  LIBRARY: 'library',
+  LEVEL: 'level'
+};
+
 export default class SpriteUpload extends React.Component {
   state = {
     fileData: null,
     filePreviewURL: '',
     filename: '',
+    spriteAvailability: '',
     category: '',
+    currentCategories: [],
     uploadStatus: {
       success: null,
       message: ''
     }
   };
 
+  componentDidMount() {
+    fetch(`/api/v1/animation-library/manifest/spritelab/en_us`)
```

Comment:
i think we should move these API calls into a separate file (e.g., `animationLibraryApi.js`) so they can be reused elsewhere. there's only one other place we call this API (in [P5LabView.jsx](https://github.com/code-dot-org/code-dot-org/blob/6ea477d3a1480dfd242df46f980428fe9b50b097/apps/src/p5lab/P5LabView.jsx#L60)), so moving the 2 requests here and that one would get us into a good spot!

Reply:
Alright, I took a first shot at this. Would love feedback on it!